### ISSUE TITLE
[no-invalid-void-type] allowInGenericTypeArguments not working with Promise.all<void>

### ISSUE THREAD

**Issue by jer-sen** [typescript-eslint/typescript-eslint#3661]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-invalid-void-type": ["warn", { "allowInGenericTypeArguments": ["Promise", "Promise.all"] }]
  }
}
```

```TS
await Promise.all<void>(voidPromisesArray);
```

**Expected Result**

No warning.

**Actual Result**

Rule warning is reported.

**Additional Info**

No.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.5` |
| `@typescript-eslint/parser`        | `4.28.5` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.31.0` |
| `node`                             | `v14.15.5` |

**Comment by bradzacher** [comment]
`allowInGenericTypeArguments` *only* applies to usage in a **generic type's arguments**.
`type T = Promise<void>`.

`Promise.all<void>()` is not a generic type - it is a function call with generic arguments.

The docs might be a little unclear on this point:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-void-type.md#allowingenerictypearguments
